src/Language/Types.hs

```diff
@@ -33,11 +33,10 @@ data InputDef = InputDef
 -- | Atomic types
 data Btype = Booltype      -- ^ Boolean
            | Itype         -- ^ Int
-           | AnySymbol     -- ^ this is the type all symbols live in
            | Input         -- ^ The input type specified at the top of the program
            | Board         -- ^ A game board
            | Top           -- ^ Really this is bottom FIXME
-           | Undef         -- ^ Not definable by a user (only occurs when typechecking)
+           | Named String
```

Comment:
this was the least invasive way I could think to introduce type names so that, for example, `f : Player - > Player` could be stored rather than `f : {A, B} -> {A, B}`